apps/nextjs/src/components/ContextProviders/ChatProvider.tsx

```diff
@@ -393,61 +455,65 @@ export function ChatProvider({ id, children }: Readonly<ChatProviderProps>) {
 
   const value: ChatContextProps = useMemo(
     () => ({
+      ailaStreamingStatus,
+      append,
+      chat,
+      chatAreaRef,
+      executeQueuedAction,
+      hasAppendedInitialMessage,
+      hasFinished,
       id,
-      chat: chat ?? undefined,
       initialModerations: moderations ?? [],
-      toxicModeration,
-      lessonPlan: overrideLessonPlan ?? tempLessonPlan,
-      hasFinished,
-      hasAppendedInitialMessage,
-      chatAreaRef,
-      append,
-      messages,
-      ailaStreamingStatus,
+      input,
       isLoading,
       isStreaming: !hasFinished,
+      iteration,
       lastModeration,
-      reload,
-      stop,
-      input,
-      setInput,
-      partialPatches,
-      validPatches,
+      lessonPlan: overrideLessonPlan ?? lessonPlan,
+      messages,
       queuedUserAction,
       queueUserAction,
-      executeQueuedAction,
+      reload,
+      setInput,
+      stop,
+      streamingSection,
+      streamingSections,
+      streamingSectionCompleted,
+      toxicModeration,
+      sectionRefs,
+      setSectionRef,
     }),
     [
-      id,
+      ailaStreamingStatus,
+      append,
       chat,
-      moderations,
-      toxicModeration,
-      tempLessonPlan,
-      hasFinished,
-      hasAppendedInitialMessage,
       chatAreaRef,
-      messages,
-      ailaStreamingStatus,
+      executeQueuedAction,
+      hasAppendedInitialMessage,
+      hasFinished,
+      id,
+      input,
       isLoading,
+      iteration,
       lastModeration,
-      reload,
-      stop,
-      input,
-      setInput,
-      append,
-      partialPatches,
-      validPatches,
+      lessonPlan,
+      messages,
+      moderations,
       overrideLessonPlan,
       queuedUserAction,
       queueUserAction,
-      executeQueuedAction,
+      reload,
+      setInput,
+      stop,
+      streamingSection,
+      streamingSections,
+      streamingSectionCompleted,
+      toxicModeration,
+      sectionRefs,
+      setSectionRef,
```

Comment:
I think this file is quickly becoming the new "central lesson plan file which is too complicated to understand" file!